File: HotSpotPrototype/Hotspot.cs
             
namespace HotSpotPrototype
{
    public class HotspotCls
    {
        private long _id;

        public string ShapeType { get; set; }
        public int[] Coordinates { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long ID { get { return _id; } set { _id = value; } }

        public HotspotCls(string shapeType, int[] coords, int width, int height, string name, string desc)
        {
            ShapeType = shapeType;
            Coordinates = coords;
            Width = width;
            Height = height;
            Name = name;
            Description = desc;
        }

        public HotspotCls()
        {
            ShapeType = "";
            Coordinates = new int[2];
            Width = 0;
            Height = 0;
            Name = "";
            Description = "";
            _id = 0;
        }

        public HotspotCls(long id)
        {
            // TODO: Get hotspot data from database.
            // get hotspot with id from data
        }
    }
}

File: HotSpotPrototype/js/Hotspot.js
function Hotspot(shape) {
    this.shape = (shape || null);
    this.id = 0;
    this.shapeType = '';
    this.coordinates = new Array();
    this.width = 0;
    this.height = 0;
    this.name = 'Test';
    this.description = 'This is a test.';
}

Hotspot.prototype.toJSON = function (includeId) {
    if (!includeId) includeId = false;

    var jstr = "{";

    if (includeId == true) {
        jstr += "'id':'" + this.id + "',";
    }

    jstr += "'shapeType':'" + this.shapeType + "'," +
                            "'coordinates':[" + this.coordinates + "]," +
                            "'width':" + this.width + "," +
                            "'height':" + this.height + "," +
                            "'name':'" + this.name + "'," +
                            "'description':'" + this.description + "'}";
    return jstr;
};